### PR TITLE
Adding "select/deselect all" checkbox

### PR DESCRIPTION
**Pull request by siempredelao**
Adding an option to select and deselect all views. This behaviour is particularly useful when the amount of views is big and we only need to inject few of them.

### CODE REVIEW

**Comment by siempredelao** [comment]
@TomasKypta any feedback would be appreciated :smile:

File: src/com/avast/android/butterknifezelezny/form/EntryHeader.java [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
     protected JLabel mName;
+    protected OnCheckBoxStateChangedListener mAllListener;
+
+    public void setmAllListener(final OnCheckBoxStateChangedListener onStateChangedListener) {
+        this.mAllListener = onStateChangedListener;
+    }
 
     public EntryHeader() {
+        mAllCheck = new JCheckBox();
+        mAllCheck.setPreferredSize(new Dimension(20, 26));
```
Comment (TomasKypta) [inline src/com/avast/android/butterknifezelezny/form/EntryHeader.java]: Wrong dimension, the checkbox is cut off. Please set the same value as for entries' checkboxes (40, 26).

File: src/com/avast/android/butterknifezelezny/form/EntryHeader.java [hunk 2]
```diff
@@ -28,7 +43,9 @@ public EntryHeader() {
         mName.setFont(new Font(mName.getFont().getFontName(), Font.BOLD, mName.getFont().getSize()));
 
         setLayout(new BoxLayout(this, BoxLayout.LINE_AXIS));
-        add(Box.createRigidArea(new Dimension(52, 0)));
+        add(Box.createRigidArea(new Dimension(1, 0)));
+        add(mAllCheck);
+        add(Box.createRigidArea(new Dimension(31, 0)));
```
Comment (TomasKypta) [inline src/com/avast/android/butterknifezelezny/form/EntryHeader.java]: Too wide, make it (11, 0)

File: src/com/avast/android/butterknifezelezny/form/Entry.java [hunk 3]
```diff
@@ -25,6 +27,14 @@
     protected Color mNameDefaultColor;
     protected Color mNameErrorColor = new Color(0x880000);
 
+    public JCheckBox getmCheck() {
```
Comment (TomasKypta) [inline src/com/avast/android/butterknifezelezny/form/Entry.java]: Please remove 'm' from all generated getters and setters.
The member fields are supposed to have m- prefix but not the getters and setters.
E.g. this method should be called getCheck()

**Comment by TomasKypta** [comment]
Hi, I found only few form size and codestyle bugs.
Otherwise the code is fine.
Could you please fix those few things?

**Comment by siempredelao** [comment]
Sure, I'll do it as soon as possible. :smile:

**Comment by siempredelao** [comment]
Hi @TomasKypta please, take a look again, I have applied your comments :wink:

**Comment by TomasKypta** [comment]
Merged. Thank you for your contribution.

**Pull request merged**